### ISSUE TITLE
ArrayIndexOutOfBoundsException if query has a single or double quote char in vendor specific comment

### ISSUE THREAD

**Issue by sung101** [jOOQ/jOOQ#9648]
### Expected behavior and actual behavior:
The query executed is as follows:

```sql
SELECT 1+1 as A from dual // This is David's comment
```

The comment should be ignored and the query should produces a result '2'.
Instead, it fails with ArrayIndexOutOfBoundsException, which produces the following stacktrace:

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: 52
	at org.jooq_3.12.3.DEFAULT.debug(Unknown Source)
	at org.jooq.impl.Tools.renderAndBind(Tools.java:2270)
	at org.jooq.impl.SQLImpl.accept(SQLImpl.java:66)
	at org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:583)
	at org.jooq.impl.AbstractContext.visit0(AbstractContext.java:457)
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:218)
	at org.jooq.impl.SQLResultQuery.accept(SQLResultQuery.java:77)
	at org.jooq.impl.DefaultRenderContext.visit0(DefaultRenderContext.java:583)
	at org.jooq.impl.AbstractContext.visit0(AbstractContext.java:457)
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:218)
	at org.jooq.impl.AbstractQuery.getSQL0(AbstractQuery.java:520)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:323)
	at org.jooq.impl.AbstractResultQuery.fetchLazy(AbstractResultQuery.java:501)
```

### Steps to reproduce the problem (if possible, create an MCVE: https://github.com/jOOQ/jOOQ-mcve):

Execute a query in Snowflake:

```sql
SELECT 1+1 as A from dual // This is David's comment
```

It will fail with ArrayIndexOutOfBoundsException before the request is sent to the db server.

### Versions:

- jOOQ:  3.12.3 (groupId: org.jooq.trial)
           Also the same error was found in:
                3.9.1 (groupId: org.jooq.pro-java-6)

- Java:  8
- Database (include vendor):  Snowflake
- OS: Linux & MacOS
- JDBC Driver (include name if inofficial driver):
snowflake_jdbc.jar

**Comment by lukaseder** [comment]
Thank you very much for your report.

We currently do not officially support Snowflake. I have created an issue to collect traction by the community, to see if we should add support in the future: #9650.

This is the first time I've seen a SQL database support c-style single line comment syntax - but why not. The c-style multi line comment syntax `/* ... */` is supported by most.

There are two issues here:

- First off, we should not throw an AIOOBE, of course. That's an easy to fix bug.
- The second issue is the fact that we don't recognise this comment syntax, which will need more thought. I haven't found an `//` operator in PostgreSQL 12, but there might be some dialect that supports this as an operator. We'll look into this separately through: #9651

**Comment by lukaseder** [comment]
As a workaround, I suggest using standard SQL comment tokens:

```sql
SELECT 1+1 as A from dual -- This is David's comment
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.13.0. Backport scheduled for jOOQ 3.12.4 (#9652)
